server/api/controllers/users/view-user.js

```diff
@@ -0,0 +1,41 @@
+module.exports = {
+  friendlyName: 'View users',
+
+  description: '',
+
+  inputs: {
+    userId: {
+      required: false,
+      type: 'string'
+    }
+  },
+
+  exits: {
+    success: {
+      description: "User account returned successfully",
+    },
+    invalid : {
+      description: "Failed to retrieve user account",
+    }
+  },
+  fn: async function({ userId }, exits) {
+    if (userId) {
+      try {
```

Comment:
can refactor this into just 1 try/catch block, maybe to something like 
```
try {
  if (userId) {
    get user id
    if (data.length === 0) throw('user not found')
  }
  
    get all users
    if (data.length === 0) throw('no users')
} catch (err) {
  handle_error
}
```

Reply:
actually now that i think about it, when the admin wants to find a user, they wouldn't use the userId. they would use emails or full names. so this query need to be adjusted to accommodate that right?